crates/pipeline_manager/src/db/service.rs

```diff
@@ -228,3 +254,85 @@ pub(crate) async fn delete_service(
         })
     }
 }
+
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn new_service_probe(
+    db: &ProjectDB,
+    tenant_id: TenantId,
+    service_id: ServiceId,
+    id: Uuid,
+    success: &bool,
+    log: &str,
+    started_at: &DateTime<Utc>,
+    finished_at: &DateTime<Utc>,
+    txn: Option<&Transaction<'_>>,
```

Comment:
Is there a reason this is a transactional API?

Reply:
In general, the transaction is there if a service name first has to be resolved to its identifier.